contracts/os/andromeda-adodb/src/state.rs

```diff
@@ -27,28 +28,35 @@ pub fn store_code_id(
 ) -> Result<(), ContractError> {
     let curr_type = ADO_TYPE.may_load(storage, &code_id.to_string())?;
     ensure!(
-        curr_type.is_none() || &curr_type.unwrap() == ado_version,
+        curr_type.is_none()
+            || &curr_type.unwrap_or(ADOVersion::from_string(String::default())) == ado_version,
         ContractError::Unauthorized {}
     );
-    ADO_TYPE
-        .save(storage, &code_id.to_string(), ado_version)
-        .unwrap();
+    ADO_TYPE.save(storage, &code_id.to_string(), ado_version)?;
     let version = semver::Version::parse(&ado_version.get_version())
         .ok()
         .ok_or(ContractError::InvalidADOVersion { msg: None })?;
     let prerelease = version.pre.parse::<String>().unwrap_or_default();
+
     if prerelease.is_empty() {
-        LATEST_VERSION
-            .save(
+        let current_ado_version = LATEST_VERSION.may_load(storage, &ado_version.get_type())?;
+
+        let should_update = current_ado_version
+            .map(|(ver, _)| {
+                let current_version = semver::Version::parse(&ver).unwrap_or(Version::new(0, 0, 0));
```

Comment:
Shouldn't we be checking each element of the version (Major, Minor, Patch) for a bump and not just an overall bump?